testing/unit_tests/test_patch_filter.py

```diff
@@ -0,0 +1,91 @@
+import os
+import pytest
+import numpy as np
+import torch.backends.cudnn as cudnn
+from torch.utils.data import DataLoader
+
+from ivadomed.loader.bids_dataframe import BidsDataframe
+from ivadomed import utils as imed_utils
+from ivadomed.loader import utils as imed_loader_utils, loader as imed_loader
+from testing.unit_tests.t_utils import create_tmp_dir,  __data_testing_dir__, __tmp_dir__, download_data_testing_test_files, path_repo_root
+from testing.common_testing_util import remove_tmp_dir
+
+cudnn.benchmark = True
+
+GPU_ID = 0
+BATCH_SIZE = 1
+
+
+def setup_function():
+    create_tmp_dir()
+
+
+def _cmpt_slice(ds_loader):
+    cmpt_label = {0: 0, 1: 0}
+    for i, batch in enumerate(ds_loader):
+        for gt in batch['gt']:
+            # TODO: multi label
+            if np.any(gt.numpy()):
+                cmpt_label[1] += 1
+            else:
+                cmpt_label[0] += 1
+    print(cmpt_label)
+    return cmpt_label[0], cmpt_label[1]
+
+
+@pytest.mark.parametrize('transforms_dict', [{"CenterCrop": {"size": [128, 128], "applied_to": ["im", "gt"]}}])
+@pytest.mark.parametrize('train_lst', [['sub-rat3_ses-01_sample-data9_SEM.png']])
+@pytest.mark.parametrize('target_lst', [["_seg-axon-manual", "_seg-myelin-manual"]])
+@pytest.mark.parametrize('patch_filter_params', [
+    {"filter_empty_mask": False, "filter_empty_input":  True},
+    {"filter_empty_mask": True, "filter_empty_input": True}])
+@pytest.mark.parametrize('dataset_type', ["training", "testing"])
+def test_patch_filter(download_data_testing_test_files, transforms_dict, train_lst, target_lst, patch_filter_params,
+    dataset_type):
+
+    cuda_available, device = imed_utils.define_device(GPU_ID)
+
+    loader_params = {
+        "transforms_params": transforms_dict,
+        "data_list": train_lst,
+        "dataset_type": dataset_type,
+        "requires_undo": False,
+        "contrast_params": {"contrast_lst": ['SEM'], "balance": {}},
+        "path_data": [os.path.join(__data_testing_dir__, "microscopy_png")],
+        "bids_config": f"{path_repo_root}/ivadomed/config/config_bids.json",
+        "target_suffix": target_lst,
+        "extensions": [".png"],
+        "roi_params": {"suffix": None, "slice_filter_roi": None},
+        "model_params": {"name": "Unet", "length_2D": [32, 32], "stride_2D": [32, 32]},
+        "slice_filter_params": {"filter_empty_mask": False, "filter_empty_input": False},
+        "patch_filter_params": patch_filter_params,
+        "slice_axis": "axial",
+        "multichannel": False
```

Comment:
We had began the migration of the all keywords (KW) classes to centrally store these keyword strings for the various params to allow easier refactoring and consistency change down the road. You might want to take a look at those: https://github.com/ivadomed/ivadomed/blob/master/ivadomed/keywords.py

Reply:
I'm not sure if you mean add new keywords to the list, or to change the parameters in the tests for keywords?
If it's the latter, it may be preferable to keep the patch filter tests "consistent" with the slice filter tests and not change for keywords in a single test file.

Reply:
Hey, no worry. Test refactoring is coming down the road for JSON consolidation reasons. So no worry about it :D.